src/System.IO.Pipelines/ReadableBuffer.cs

```diff
@@ -250,14 +250,14 @@ public static ReadableBuffer Create(byte[] data, int offset, int length)
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.length);
             }
 
-            var buffer = new OwnedArray<byte>(data);
+            OwnedBuffer<byte> buffer = data;
```

Comment:
Implicit ctor via cast via declaration?

Wouldn't `var buffer = (OwnedBuffer<byte>)data;` be better or expose a `Create(T[] array)` function or people might either get confused how to create or start using (due to not working out how to create the buffer)
```
var buffer = Create(new ArraySegment<T>(array, 0, array.Length);
```
Which would be inefficent